code/modules/food_and_drinks/food/snacks.dm

```diff
@@ -119,46 +119,46 @@
 	if((slices_num <= 0 || !slices_num) || !slice_path)
 		return 0
 
-	var/inaccurate = 0
-	if( \
-			istype(W, /obj/item/kitchen/knife) || \
-			istype(W, /obj/item/scalpel) \
-		)
-	else if( \
-			istype(W, /obj/item/circular_saw) || \
-			istype(W, /obj/item/melee/energy/sword/saber) && W:active || \
-			istype(W, /obj/item/melee/energy/blade) || \
-			istype(W, /obj/item/shovel) || \
-			istype(W, /obj/item/hatchet) \
-		)
-		inaccurate = 1
-	else if(W.w_class <= WEIGHT_CLASS_SMALL && istype(src,/obj/item/reagent_containers/food/snacks/sliceable))
+	if(W.sharp) // First check if the object is sharp, we're probably trying to cut the food.
+		var/success = FALSE //Assume we fail to cut it
+		if(istype(W, /obj/item/kitchen/knife) || istype(W, /obj/item/scalpel)) // We do an accurate cut with these items, everything else is assumed inaccurate
+			success = slice(W, user, FALSE)
+		else
+			success = slice(W, user, TRUE)
+		if(success) // If we successful sliced the food then we stop here, otherwise continue to the next check
+			return
+
+	if(W.w_class <= WEIGHT_CLASS_SMALL && istype(src,/obj/item/reagent_containers/food/snacks/sliceable))
 		var/newweight = GetTotalContentsWeight() + W.GetTotalContentsWeight() + W.w_class
 		if(newweight > MAX_WEIGHT_CLASS)
 			// Nope, no bluespace slice food
 			to_chat(user, "<span class='warning'>You cannot fit [W] in [src]!</span>")
 			return 1
 		if(!iscarbon(user))
 			return 1
-		to_chat(user, "<span class='warning'>You slip [W] inside [src].</span>")
-		user.unEquip(W)
-		if((user.client && user.s_active != src))
-			user.client.screen -= W
-		W.dropped(user)
-		total_w_class += W.w_class
-		add_fingerprint(user)
-		contents += W
-		return
+		var/contine = alert("Are you sure you want to put [W] inside [src]", "Food filling", "Yes", "No")
+		if(contine == "Yes" && Adjacent(user))
+			to_chat(user, "<span class='warning'>You slip [W] inside [src].</span>")
+			user.unEquip(W)
+			if((user.client && user.s_active != src))
+				user.client.screen -= W
+			W.dropped(user)
+			total_w_class += W.w_class
+			add_fingerprint(user)
+			contents += W
+			return
 	else
 		return 1
+
+/obj/item/weapon/reagent_containers/food/snacks/proc/slice(obj/item/weapon/W, mob/user, inaccurate = FALSE)
```

Comment:
You forgot to remove `/weapon/` from the path here, so this proc isn't being found by the rest of the code (hence your TRAVIS and compile errors)